service/internal/proctelemetry/process_telemetry.go

```diff
@@ -63,18 +63,27 @@ type processMetrics struct {
 
 // RegisterProcessMetrics creates a new set of processMetrics (mem, cpu) that can be used to measure
 // basic information about this process.
-func RegisterProcessMetrics(ocRegistry *metric.Registry, mp otelmetric.MeterProvider, useOtel bool, ballastSizeBytes uint64) error {
+func RegisterProcessMetrics(ocRegistry *metric.Registry, mp otelmetric.MeterProvider, useOtel bool, ballastSizeBytes uint64, useHostProcEnvVar bool) error {
 	var err error
 	pm := &processMetrics{
 		startTimeUnixNano: time.Now().UnixNano(),
 		ballastSizeBytes:  ballastSizeBytes,
 		ms:                &runtime.MemStats{},
 	}
 
+	hostproc, exists := os.LookupEnv("HOST_PROC")
+	// unset HOST_PROC env variable so the process search occurs locally
+	if !useHostProcEnvVar && exists {
+		os.Unsetenv("HOST_PROC")
+	}
 	pm.proc, err = process.NewProcess(int32(os.Getpid()))
 	if err != nil {
 		return err
 	}
+	// restore immediately if it was previously set, don't use defer
```

Comment:
If we don't use `defer` and `process.NewProcess` panics we won't set the environment variable back, so I don't think this approach is acceptable. What are the challenges with using `defer`?

I think a better approach for this would be for `gopsutil` to allow us specifying the use or not of `HOST_PROC` instead of us modifying the environment variables. Have we asked the `gopsutil` maintainer about this?